pkg/services/pluginsintegration/plugincontext/plugincontext.go

```diff
@@ -11,68 +11,83 @@ import (
 
 	"github.com/grafana/grafana/pkg/infra/localcache"
 	"github.com/grafana/grafana/pkg/plugins"
-	"github.com/grafana/grafana/pkg/plugins/log"
 	"github.com/grafana/grafana/pkg/services/datasources"
 	"github.com/grafana/grafana/pkg/services/pluginsintegration/adapters"
 	"github.com/grafana/grafana/pkg/services/pluginsintegration/pluginsettings"
 	"github.com/grafana/grafana/pkg/services/user"
 )
 
+var ErrPluginNotFound = errors.New("plugin not found")
+
 func ProvideService(cacheService *localcache.CacheService, pluginStore plugins.Store,
-	dataSourceCache datasources.CacheService, dataSourceService datasources.DataSourceService,
-	pluginSettingsService pluginsettings.Service) *Provider {
+	dataSourceService datasources.DataSourceService, pluginSettingsService pluginsettings.Service,
+	keyProvider KeyProvider) *Provider {
 	return &Provider{
+		keyProvider:           keyProvider,
 		cacheService:          cacheService,
 		pluginStore:           pluginStore,
-		dataSourceCache:       dataSourceCache,
 		dataSourceService:     dataSourceService,
 		pluginSettingsService: pluginSettingsService,
-		logger:                log.New("plugincontext"),
 	}
 }
 
 type Provider struct {
 	cacheService          *localcache.CacheService
 	pluginStore           plugins.Store
-	dataSourceCache       datasources.CacheService
 	dataSourceService     datasources.DataSourceService
 	pluginSettingsService pluginsettings.Service
-	logger                log.Logger
+	keyProvider           KeyProvider
 }
 
 // Get allows getting plugin context by its ID. If datasourceUID is not empty string
 // then PluginContext.DataSourceInstanceSettings will be resolved and appended to
 // returned context.
-func (p *Provider) Get(ctx context.Context, pluginID string, user *user.SignedInUser) (backend.PluginContext, bool, error) {
-	return p.pluginContext(ctx, pluginID, user)
+func (p *Provider) Get(ctx context.Context, pluginID string, user *user.SignedInUser) (backend.PluginContext, error) {
+	plugin, exists := p.pluginStore.Plugin(ctx, pluginID)
+	if !exists {
+		return backend.PluginContext{}, ErrPluginNotFound
+	}
+
+	pCtx := backend.PluginContext{
+		OrgID:    user.OrgID,
+		PluginID: pluginID,
+		User:     adapters.BackendUserFromSignedInUser(user),
+	}
+
+	if plugin.IsApp() {
+		appSettings, err := p.appInstanceSettings(ctx, pluginID, user)
+		if err != nil {
+			return backend.PluginContext{}, err
+		}
+		pCtx.AppInstanceSettings = appSettings
+		//pCtx.Key = p.keyProvider.AppKey(ctx, pluginID, user.OrgID)
```

Comment:
leftover? I don't see the `keyProvider` being used. Am I missing something?

Reply:
Ditto https://github.com/grafana/grafana/pull/66451#discussion_r1165662601